WalletWasabi.Documentation/RPC.md

```diff
@@ -0,0 +1,436 @@
+
+```bash
+curl -s --data-binary '{"jsonrpc":"2.0","id":"1","method":"gethistory"}' http:/127.0.0.1:18099
+```
+
+```json
+  "jsonrpc": "2.0",
+  "result": [
+    {
+      "datetime": "2019-10-01T12:31:57+00:00",
+      "height": 597871,
+      "amount": -2110090,
+      "label": "David",
+      "tx": "680d8940145f53cf2a0c24b27ba0bb53fbd639011eba3d23c9d53123ddae5f32"
+    },
+    {
+      "datetime": "2019-10-04T17:00:15+00:00",
+      "height": 597872,
+      "amount": -2120000,
+      "label": "Pablo",
+      "tx": "e5e4486ad2c9fc6f3c262c4c64fa5fbcc607aa301182d45848a6692c5a0d0fc0"
+    },
+    {
+      "datetime": "2019-09-22T11:59:32+00:00",
+      "height": 5965600,
+      "amount": 44480000,
+      "label": "Coinbase",
+      "tx": "6a2e99298dbbd201230a99e62ea584d7f63f62ad1de7166f24eb2e24867f6faf"
+    },
+```
+
+## listkeys
+
+Returns the list of all the generated keys.
+
+```bash
+curl --data-binary '{"jsonrpc":"2.0","id":"1","method":"listkeys"}' http://127.0.0.1:18099/
+```
+
+```json
+  "jsonrpc": "2.0",
+  "result": [
+    [{
+      "fullKeyPath": "84'/0'/0'/0/1",
+      "internal": false,
+      "keyState": 1,
+      "label": "Adam",
+      "p2wpkhScript": "0 09f4ef9a012844a875222fb08fba86219d181b63",
+      "pubkey": "03d02d5985dd48880edccc992b6d77584e7b68ad9b099b8e33d60a703ac0cbded7",
+      "pubKeyHash": "09f4ef9a012844a875222fb08fba86219d181b63"
+    },
+    {
+      "fullKeyPath": "84'/0'/0'/0/2",
+      "internal": false,
+      "keyState": 1,
+      "label": "Valeria",
+      "p2wpkhScript": "0 cceac198896736d15281b499dea4d49687d2ccf5",
+      "pubkey": "0203741debe056d1513940f36d0b16a49185c1780ac8f5b91e713d5a66fbc84025",
+      "pubKeyHash": "cceac198896736d15281b499dea4d49687d2ccf5"
+    },
+    {
+      "fullKeyPath": "84'/0'/0'/1/2",
+      "internal": true,
+      "keyState": 1,
+      "label": "Bernardo",
+      "p2wpkhScript": "0 7242d77e2da9d3474190e465ed50f6b30b3cf2e9",
+      "pubkey": "03f31132f80ebdf480ad2aaf15631e9b643734ab78701012ba621b113c23ecab26",
+      "pubKeyHash": "7242d77e2da9d3474190e465ed50f6b30b3cf2e9"
+    }
+  ],
+  "id": "1"
+}
+```
+
+## enqueue / dequeue
+
+Enqueue coins in order to participate in coinjoin.
+
+### enqueue
+
+```bash
+curl -s --data-binary '{"jsonrpc":"2.0","id":"1","method":"enqueue", "params": { "coins": [{"transactionId": "ba70587b37ba8b4de143929994d3b8ee2810340cef23e8016020687716117a52", "index":"12"}]} }' http://127.0.0.1:18099/
+```
+
+### dequeue
+
+```bash
+curl -s --data-binary '{"jsonrpc":"2.0","id":"1","method":"dequeue", "params": { "coins": [{"transactionId": "ba70587b37ba8b4de143929994d3b8ee2810340cef23e8016020687716117a52", "index":"12"}]} }' http://127.0.0.1:18099/
+```
+
+## stop
+
+Stops and exit Wasabi.
+
+```bash
+curl -s --data-binary '{"jsonrpc":"2.0", "method":"stop"}' http://127.0.0.1:18099/
+```
+
+------
+
+## Errors
+
+### Method not found
+
+```bash
+$ curl -s --data-binary '{"jsonrpc":"2.0","id":"1","method":"howknows"}' http://127.0.0.1:18099/
+{
```

Comment:
```suggestion

```json
{
```

Reply:
no, the idea is to display what happens with a real command line example.